### PR TITLE
core: Replace all illegal chars in disk aliases

### PR DESCRIPTION
**Pull request by smelamud**
When importing a VM, aliases are generated for all imported disks. To
comply with oVirt naming convention for disk aliases, we need to replace
all illegal characters that may appear in imported disk names.

Change-Id: I4f54a3fd8e31799d8ca33d2f4413e82f452ebabd
Bug-Url: https://bugzilla.redhat.com/1904962
Signed-off-by: Shmuel Melamud <smelamud@redhat.com>

### CODE REVIEW

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java [hunk 1]
```diff
@@ -75,6 +75,7 @@
 
     private static final Pattern VMWARE_DISK_NAME_PATTERN = Pattern.compile("\\[.*?\\] .*/(.*).vmdk");
     private static final Pattern DISK_NAME_PATTERN = Pattern.compile(".*/([^.]+).*");
+    private static final Pattern DISK_ALIAS_ILLEGAL_CHARS_PATTERN = Pattern.compile("[^\\w.-]");
```
Comment (liranr23) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java:78]: you allow only word chars? numeric is impossible? can you tell what exactly is illegal?

Comment (smelamud) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java:78]: `\w` includes latin letters, digits and underscore. Also period (`.`) and hyphen (`-`) are allowed. These are characters that we allow for disk alias in WebAdmin.

Comment (liranr23) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java:78]: > `\w` includes latin letters, digits and underscore. Also period (`.`) and hyphen (`-`) are allowed. These are characters that we allow for disk alias in WebAdmin.

OK

**Review by liranr23 — APPROVED** [review]

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java (hunk 1 above)
Comment (ahadas) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java:78]: > `\w` includes latin letters, digits and underscore. Also period (`.`) and hyphen (`-`) are allowed. These are characters that we allow for disk alias in WebAdmin.

so can we reuse the pattern that is used by the webadmin?

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java [hunk 2]
```diff
@@ -369,7 +370,7 @@ && getStorageDomain().getStorageType().isBlockDomain()) {
     }
 
     private static String replaceInvalidDiskAliasChars(String alias) {
-        return alias.replace(' ', '_');
+        return DISK_ALIAS_ILLEGAL_CHARS_PATTERN.matcher(alias).replaceAll("_");
```
Comment (ahadas) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/ImportVmFromExternalProviderCommand.java:373]: how about adding a unit test?

**Comment by smelamud** [comment]
Verified.

**Review by ahadas — APPROVED** [review]

**Pull request merged**